src/test/java/org/folio/edge/controller/InnReachProxyControllerTest.java

```diff
@@ -98,4 +107,72 @@ void shouldUpdateRequestHeaders() {
     );
   }
 
+  @Test
+  void shouldHandleCommonErrors() {
+    var httpHeaders = createInnReachHttpHeaders();
+    httpHeaders.set(ACCEPT_ENCODING, "gzip");
+    httpHeaders.set(AUTHORIZATION, AUTH_TOKEN_VALUE);
+
+    InnReachResponseDTO innReachResponseDTO = new InnReachResponseDTO();
+    innReachResponseDTO.status("failed");
+    ObjectMapper obj = new ObjectMapper();
+    String jsonStr = null;
+    try {
+      jsonStr = obj.writeValueAsString(innReachResponseDTO);
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    HttpEntity<InnReachResponseDTO> requestEntity = new HttpEntity<>(innReachResponseDTO, httpHeaders);
+
+    wireMock.stubFor(post(PATRON_VERIFY_URL_PATTERN)
+      .willReturn(aResponse().withBody(jsonStr)
+        .withStatus(400)
+        .withHeader(CONTENT_TYPE, APPLICATION_JSON_VALUE)
+        .withHeader(X_OKAPI_TOKEN, TEST_TOKEN)));
+
+    ResponseEntity<InnReachResponseDTO> responseEntity = testRestTemplate.exchange(BASE_URI + "/circ/verifypatron",
+      POST, requestEntity, InnReachResponseDTO.class);
+
+    assertTrue(responseEntity.getStatusCode().is4xxClientError());
+    assertEquals(BAD_REQUEST, responseEntity.getStatusCode());
+    assertTrue(responseEntity.hasBody());
+    assertEquals(innReachResponseDTO, responseEntity.getBody());
+  }
+
+  @Test
+  void shouldHandleCommonErrorsWhenCantReadBody() {
+    var httpHeaders = createInnReachHttpHeaders();
+    httpHeaders.set(ACCEPT_ENCODING, "gzip");
+    httpHeaders.set(AUTHORIZATION, AUTH_TOKEN_VALUE);
+
+    InnReachResponseDTO innReachResponseDTO = new InnReachResponseDTO();
+    innReachResponseDTO.status("failed");
+    InnReachRequest innReachRequest = new InnReachRequest();
+    ObjectMapper obj = new ObjectMapper();
+    String jsonStr = null;
+    try {
+      jsonStr = obj.writeValueAsString(innReachRequest);
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    HttpEntity<InnReachResponseDTO> requestEntity = new HttpEntity<>(innReachResponseDTO, httpHeaders);
+
+    wireMock.stubFor(post(PATRON_VERIFY_URL_PATTERN)
+      .willReturn(aResponse().withBody(jsonStr)
+        .withStatus(400)
+        .withHeader(CONTENT_TYPE, APPLICATION_JSON_VALUE)
+        .withHeader(X_OKAPI_TOKEN, TEST_TOKEN)));
```

Comment:
As I understand, it should verify the fall-back scenario when json can't be serialized. Why do we need `innReachResponseDTO` and `innReachRequest` objects for this? Simplify it to something like:
```
wireMock.stubFor(post(PATRON_VERIFY_URL_PATTERN)
      .willReturn(aResponse().withBody("Some plain text")
        .withStatus(400)
        .withHeader(CONTENT_TYPE, TEXT_PLAIN_VALUE)
        .withHeader(X_OKAPI_TOKEN, TEST_TOKEN))); 
```